src/actions/setup-challenge.ts

```diff
@@ -84,11 +91,39 @@ export const setupChallenge = async (name: string, installLocation: string) => {
         console.log(chalk.green("Challenge setup completed successfully."));
         console.log("");
         console.log(chalk.cyan(`Now open this repository in your favorite code editor and look at the readme for instructions: ${targetDir}`));
-    } catch (error) {
-        console.error(chalk.red("An error occurred during challenge setup:"), error);
+    } catch (error: any) {
+        console.error(chalk.red("An error occurred during challenge setup:"), error.message);
+    }
+}
+
+const checkDependencyInstalled = async (name: RequiredDependency) => {
+    try {
+        await execa(name, ["--help"]);
+    } catch(_) {
+        throw new Error(`${name} is required. Please install to continue.`);
     }
 }
 
+const checkDependencyVersion = async (name: RequiredDependency, requiredVersion: string | Range) => {
+    try {
+        const userVersion = (await execa(name, ["--version"])).stdout;
+        if (!semver.satisfies(userVersion, requiredVersion)) {
+            throw new Error(`${name} version requirement of ${requiredVersion} not met. Please update to continue.`);
+        }
+    } catch(_) {
+        throw new Error(`${name} ${requiredVersion} is required. Please install to continue.`);
```

Comment:
This just notifies the user that their missing a dependency. Another feature we could potentially add is to then give them a prompt to download the dependency, all within the context of the CLI

Reply:
That would be very fancy. I think what you have is good for now and if you ever felt inspired to go that path you could make another PR.